src/uattributes/uattributesvalidator.rs

```diff
@@ -86,14 +86,14 @@ pub trait UAttributesValidator {
     /// * the current system time cannot be determined.
     fn is_expired(&self, attributes: &UAttributes) -> Result<(), UAttributesError> {
         let ttl = match attributes.ttl {
-            Some(t) if t > 0 => {
-                // this will not be needed anymore once up-core-api 1.5.7 is released and
-                // ttl is defined as a uint32
-                u64::try_from(t).expect("should have been able to convert positive i32 to u64")
-            }
+            Some(t) => u64::from(t),
```

Comment:
IMHO we should change this to
```rust
Some(t) if t > 0 => u64::from(t),
_ => return Ok(()),
```

and then remove the block below
```rust
if ttl == 0 {
  return Ok(());
}
```

Reply:
nice